catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java

```diff
@@ -74,539 +69,164 @@
-                    errorMessage.append("\n");
-                }
-
-                // If warnings are to be included from the Schematron report as part of the errors
-                // message
-                if (!this.suppressWarnings) {
-                    List<String> warnings = this.report.getWarnings();
-                    LOGGER.debug("warnings.size() = {}", warnings.size());
-                    for (String warning : warnings) {
-                        LOGGER.debug("warning = {}", warning);
-                        errorMessage.append(warning);
-                        errorMessage.append("\n");
-                    }
-
-                }
-                throw new StopProcessingException(errorMessage.toString());
-            }
-
-        } catch (TransformerException te) {
-            LOGGER.debug("Unable to setup validator", te);
-            LOGGER.debug("EXITING: {}.{}", CLASS_NAME, methodName);
-
-            throw new StopProcessingException("Could not setup validator to perform validation.");
-        }
-
-        LOGGER.debug("EXITING: {}.{}", CLASS_NAME, methodName);
-    }
-
-    /**
-     * Retrieve the Schematron validation results.
-     *
-     * @return Schematron validation output report
-     */
-    public SchematronReport getSchematronReport() {
-        return this.report;
-    }
-
-    /**
-     * Retrieve the name of the original Schematron .sch file provided as input to this validation
-     * service.
-     *
-     * @return name of original Schematron .sch file
-     */
-    public String getSchematronSchemaFilename() {
-        return this.schematronSchemaFilename;
-    }
-
-    /**
-     * Retrieve suppress warnings flag.
-     *
-     * @return true indicates Schematron warnings are being suppressed
-     */
-    public boolean getSuppressWarnings() {
-        LOGGER.debug("ENTERING: getSuppressWarnings");
-
-        return this.suppressWarnings;
-    }
-
-    /**
-     * Suppress Schematron warnings, such that only errors mark a request as invalid. This method is
-     * called whenever the Save button is selected for a Schematron ruleset bundle on the OSGi
-     * container's web console Configuration admin page.
-     *
-     * @param suppressWarnings
-     *            true indicates Schematron warnings are to be suppressed
-     */
     public void setSuppressWarnings(boolean suppressWarnings) {
-        LOGGER.debug("ENTERING: setSuppressWarnings");
-        LOGGER.debug("suppressWarnings = {} (sch filename = {})", suppressWarnings,
-                this.schematronSchemaFilename);
-
         this.suppressWarnings = suppressWarnings;
+    }
 
-        LOGGER.debug("EXITING: setSuppressWarnings");
+    public void setSchematronFileNames(ArrayList<String> schematronFileNames) {
+        this.schematronFileNames = new ArrayList<>();
+        for (String file : schematronFileNames) {
+            String fullpath = (file.startsWith("/")) ? file : SCHEMATRON_BASE_FOLDER + file;
```

Comment:
This won't work on Windows. Use Java Paths instead https://docs.oracle.com/javase/8/docs/api/java/nio/file/Paths.html